context.go

```diff
@@ -24,18 +24,51 @@ import (
 	gcontext "context"
 
 	"go.uber.org/fx/service"
+	"go.uber.org/fx/ulog"
 )
 
+var _logger = "logger"
+
 // Context embeds Host and Go stdlib context for use
 type Context interface {
 	gcontext.Context
-	service.Host
+
+	// unexported func ensures only fx package implements Context
+	fx()
+	Logger() ulog.Log
 	WithContext(ctx gcontext.Context) *context
 }
 
 type context struct {
 	gcontext.Context
-	service.Host
+}
+
+// Convert context.Context to fx.Context
+func Convert(ctx gcontext.Context) Context {
+	fx, ok := ctx.(Context)
+	if ok {
+		return fx
+	}
```

Comment:
What do you think about type assertion before wrapping? I am not sure of the performance overhead of a check vs additional allocation.